src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -236,6 +242,14 @@ void DlgPrefWaveform::slotSetGridLines(bool displayGrid) {
     WaveformWidgetFactory::instance()->setDisplayBeatGrid(displayGrid);
 }
 
+void DlgPrefWaveform::slotSetGridLinesAlpha(bool alphaGrid) {
+    if (m_pConfig) {
+        m_pConfig->set(ConfigKey("[Waveform]", "dimModeCheckBox"), ConfigValue(alphaGrid));
```

Comment:
you can use here the template implementation 
m_pConfig->setValue(ConfigKey("[Waveform]", "dimModeCheckBox"), alphaGrid);